src/Compilers/Core/Portable/CodeAnalysisResources.resx

```diff
@@ -450,7 +450,7 @@
     <value>Reported diagnostic has an ID '{0}', which is not a valid identifier.</value>
   </data>
   <data name="InvalidDiagnosticLocationReported" xml:space="preserve">
-    <value>Reported diagnostic '{0}' has a source location in file '{1}', which is not part of the compilation being analyzed.</value>
+    <value>Reported diagnostic '{0}' has a source location {1} in file '{2}', which is not part of the compilation being analyzed.</value>
```

Comment:
>Reported diagnostic '{0}' has a source location {1} in file '{2}', which is not part of the compilation being analyzed. [](start = 11, length = 119)

Given the new condition, I believe the message could be misleading now. Consider reflecting the span requirement in the message.